### PR TITLE
test: improve video snapshots

### PR DESCRIPTION
**Pull request by colincclark**
- improve snaps
- update README
- add test iterator
- refactor video in line with other packages

### CODE REVIEW

File: packages/video/__tests__/play-icon-shared.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+export default () => {
+  it("play icon", () => {
+    addSerializers(
+      expect,
+      enzymeTreeSerializer(),
+      compose(print, minimaliseTransform((value, key) => key === "style"))
+    );
+
+    const wrapper = shallow(<PlayIcon />);
```
Comment (craigbilner) [inline packages/video/__tests__/play-icon-shared.js]: should we be testing private/partial functionality? From a snapshot point of view we have all the tools to confirm it's "contextually correct". Either we should do a full render or we should pull it out of one with enzyme/rtr?

File: packages/video/__tests__/android/__snapshots__/video.android.test.js.snap [hunk 2]
```diff
@@ -48,215 +13,73 @@ exports[`Video on native renders correctly 1`] = `
       }
     }
   >
-    <View
+    <TimesImage
```
Comment (craigbilner) [inline packages/video/__tests__/android/__snapshots__/video.android.test.js.snap]: I think there's a lot of value in separating out the style tests which reduce/focus PR diffs when only styling changes are made?

File: packages/video/__tests__/web/__snapshots__/video-error-with-style.web.test.js.snap [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    "backgroundColor": "rgba(0, 0, 0, 0.6)",
+    "display": "flex",
+    "height": 100,
+    "justifyContent": "center",
+    "width": 100,
+  },
+}
+</style>
+
+<View
```
Comment (craigbilner) [inline packages/video/__tests__/web/__snapshots__/video-error-with-style.web.test.js.snap]: remove native elements

File: packages/video/__tests__/web/__snapshots__/video-with-provider.web.test.js.snap [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+        src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
+      />
+      <div>
+        <div />
+      </div>
+    </div>
+  </div>
+  <div>
+    <div
+      dir="auto"
```
Comment (craigbilner) [inline packages/video/__tests__/web/__snapshots__/video-with-provider.web.test.js.snap]: `minimalWeb` will clean this up

File: packages/video/__tests__/web/__snapshots__/video-with-provider.web.test.js.snap [hunk 5]
```diff
@@ -0,0 +1,78 @@
+// Jest Snapshot v1, https://goo.gl/fbAQLP
+
+exports[`1. paidonly video for unpaid users 1`] = `
+<div>
+  <div>
+    <div>
+      <img
+        alt=""
```
Comment (craigbilner) [inline packages/video/__tests__/web/__snapshots__/video-with-provider.web.test.js.snap]: we should probably be erroring on things like this?

File: packages/video/__tests__/web/is-paid.web.test.js [hunk 6]
```diff
@@ -0,0 +1,12 @@
+import { isPaidOnly } from "../../src/video";
+
+it("determine isPaidOnly status", () => {
+  expect(isPaidOnly(true)).toBe(true);
```
Comment (craigbilner) [inline packages/video/__tests__/web/is-paid.web.test.js]: if you _have_ to make multiple expectations, jest doesn't have the concept of custom messages so would ideally have these as individual tests

File: packages/video/__tests__/web/__snapshots__/video-with-provider.web.test.js.snap (hunk 4 above)
Comment (colincclark) [reply packages/video/__tests__/web/__snapshots__/video-with-provider.web.test.js.snap]: done

File: packages/video/__tests__/web/__snapshots__/video-error-with-style.web.test.js.snap (hunk 3 above)
Comment (colincclark) [reply packages/video/__tests__/web/__snapshots__/video-error-with-style.web.test.js.snap]: done

File: packages/video/__tests__/web/__snapshots__/video-with-provider.web.test.js.snap (hunk 5 above)
Comment (colincclark) [reply packages/video/__tests__/web/__snapshots__/video-with-provider.web.test.js.snap]: https://github.com/newsuk/times-components/issues/1108 also mocked out the Image component, as that should be tested in the image package

File: packages/video/__tests__/web/is-paid.web.test.js (hunk 6 above)
Comment (colincclark) [reply packages/video/__tests__/web/is-paid.web.test.js]: done, have split these out

File: packages/video/__tests__/android/__snapshots__/video.android.test.js.snap (hunk 2 above)
Comment (colincclark) [reply packages/video/__tests__/android/__snapshots__/video.android.test.js.snap]: done

File: packages/video/__tests__/play-icon-shared.js (hunk 1 above)
Comment (colincclark) [reply packages/video/__tests__/play-icon-shared.js]: done

File: packages/video/__tests__/android/__snapshots__/video-with-style.android.test.js.snap [hunk 7]
```diff
@@ -0,0 +1,85 @@
+// Jest Snapshot v1, https://goo.gl/fbAQLP
+
+exports[`1. video 1`] = `
+<Touchable
+  accessibilityLabel="splash-component"
```
Comment (craigbilner) [inline packages/video/__tests__/android/__snapshots__/video-with-style.android.test.js.snap]: optional: these would be removed

**Pull request merged**